File: app/src/main/java/com/example/apka_ksp/MainActivity.java
package com.example.apka_ksp;

import androidx.appcompat.app.AppCompatActivity;
import com.anychart.AnyChart;
import com.anychart.AnyChartView;
import com.anychart.chart.common.dataentry.DataEntry;
import com.anychart.chart.common.dataentry.ValueDataEntry;
import com.anychart.charts.LinearGauge;
import com.github.nkzawa.emitter.Emitter;
import com.github.nkzawa.socketio.client.IO;
import com.github.nkzawa.socketio.client.Socket;

import org.json.JSONException;
import org.json.JSONObject;

import java.net.URISyntaxException;
import java.util.ArrayList;
import java.util.List;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;


public class MainActivity extends AppCompatActivity {
Button btn;
TextView temp;


private Socket socket;

    {
        try {
            socket = IO.socket("https://ksp-project.azurewebsites.net/");
        } catch (URISyntaxException e) {
            e.printStackTrace();
        }
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        socket.connect();
        socket.emit("confirmation", true);
        btn = (Button)findViewById(R.id.button2) ;
        //nickname = (EditText) findViewById(R.id.nickname);
        temp = (TextView) findViewById(R.id.tempid);


        socket.on("currentValue", new Emitter.Listener() {
            @Override
            public void call(final Object... args) {
                runOnUiThread(new Runnable() {
                    @Override
                    public void run() {
                        JSONObject data = (JSONObject) args[0];
                        String value;

                        try {
                            value = data.getString("value");
                            Log.i("i", value);
                        } catch (JSONException e) {
                            return;
                        }
                        //updateText(value);
                        temp.setText(value + "\u2103");

                    }
                });
            }
        });

        btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                    Intent i  = new Intent(MainActivity.this,Chart.class);
                    startActivity(i);
                }
            }
        );
    }

}

File: app/src/main/java/com/example/apka_ksp/MainActivity.kt
package com.example.apka_ksp

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import com.example.apka_ksp.SocketIO.Companion.getValueFromJSON
import com.github.nkzawa.socketio.client.Socket
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : Activity() {
    private var socket: Socket? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        buttonChart.setOnClickListener {
            if (socket == null) {
                Toast.makeText(this, "You are not connected", Toast.LENGTH_SHORT).show()
            } else {
                openChartActivity()
            }
        }
        buttonConnect.setOnClickListener {
            if (buttonConnect.text == "Connect") {
                connectToServer()
            } else {
                socket?.disconnect()
            }
        }
    }

    private fun openChartActivity() {
        val intent = Intent(this, ChartActivity::class.java)
        startActivity(intent)
    }

    private fun connectToServer() {
        socket = SocketIO.instance.createSocket(editTextAddress.text.toString())
        if (socket == null) {
            Toast.makeText(this, "Cannot connect", Toast.LENGTH_SHORT).show()
        } else {
            socket?.connect()
            socket?.on(Socket.EVENT_CONNECT) {
                onEventConnect()
            }?.on(Socket.EVENT_DISCONNECT) {
                onEventDisconnect()
            }
        }
    }

    private fun onEventDisconnect() {
        runOnUiThread {
            Toast.makeText(this, "Disconnected", Toast.LENGTH_SHORT).show()
            buttonConnect.text = "Connect"
            buttonChart.isEnabled = false
        }
        SocketIO.instance.listOfValues.clear()
    }

    private fun onEventConnect() {
        //TODO chyba tu trzeba wrzucić create room ale dodało mi kilka pokoi na raz :/
        // socket?.emit("createRoom", "roomName")
        runOnUiThread {
            Toast.makeText(this, "Connected", Toast.LENGTH_SHORT).show()
            buttonConnect.text = "Disconnect"
            buttonChart.isEnabled = true
        }
        addCurrentValueListener()
    }

    private fun addCurrentValueListener() {
        socket?.on("currentValue") { args ->
            socket?.emit("confirmation", true)
            val value = getValueFromJSON(args)
            SocketIO.instance.listOfValues.add(value)
            drawValue(value)
        }
    }

    private fun drawValue(value: Double) {
        runOnUiThread {
            if (isWhole(value)) {
                textViewTemp.text = "${value.toInt()}℃"
            } else {
                textViewTemp.text = "$value℃"
            }
        }
    }

    private fun isWhole(value: Double) = value - value.toInt() == 0.0
}